src/org/labkey/test/tests/PermissionsTestForJavascriptExecution.java

```diff
@@ -0,0 +1,139 @@
+package org.labkey.test.tests;
+
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.labkey.test.BaseWebDriverTest;
+import org.labkey.test.Locator;
+import org.labkey.test.categories.Daily;
+import org.labkey.test.pages.query.QueryMetadataEditorPage;
+import org.labkey.test.pages.query.SourceQueryPage;
+import org.labkey.test.util.ApiPermissionsHelper;
+import org.labkey.test.util.PermissionsHelper;
+
+import java.util.Arrays;
+import java.util.List;
+
+@Category({Daily.class})
+public class PermissionsTestForJavascriptExecution extends BaseWebDriverTest
+{
+    private static final String USER = "javascripttestuser@permissionstestforjavascriptexecution.test";
+    private static final String XML_METADATA_2 = "<tables xmlns=\"http://labkey.org/data/xml\"> \n" +
```

Comment:
These are great candidates for Java text blocks. IntelliJ should offer to auto-refactor.

https://www.baeldung.com/java-text-blocks

Reply:
Done